universal-login-relayer/test/integration/ethereum/WalletContractService.test.ts

```diff
@@ -36,6 +41,47 @@ describe('INT: WalletContractService', () => {
       refundReceiver: TEST_ACCOUNT_ADDRESS,
     };
     signedMessage = messageToSignedMessage(message, wallet.privateKey, 'istanbul', 'beta2');
+    fetchWalletVersionSpy = sinon.spy((walletContractService as any).blockchainService, 'fetchWalletVersion');
+  });
+
+  describe('version cache', () => {
+    it('undefined address', async () => {
+      await expect(walletContractService.getWalletVersion(undefined as any)).to.be.rejectedWith('Invalid address provided');
+    });
+
+    it('empty address', async () => {
+      await expect(walletContractService.getWalletVersion('')).to.be.rejectedWith('Invalid address provided');
+    });
+
+    it('clear cache', async () => {
+      await walletContractService.getWalletVersion(proxyContract.address);
+      await walletContractService.clearCatch();
+      expect((walletContractService as any).walletVersions).to.deep.eq({});
+    });
+
+    it('address with empty cache', async () => {
+      const version = await walletContractService.getWalletVersion(proxyContract.address);
+      expect(version).to.eq('beta2');
+      expect(fetchWalletVersionSpy).to.be.calledOnce;
+    });
+
+    it('adding cache after calling address with empty cache', async () => {
+      await walletContractService.getWalletVersion(proxyContract.address);
+      expect((walletContractService as any).walletVersions).to.deep.eq({[proxyContract.address]: 'beta2'});
+      expect(fetchWalletVersionSpy).to.be.calledOnce;
+    });
```

Comment:
You can merge these two in one test `returns proper wallet version and saves it to the cache`